src/main/scala/symsim/ValueFunction.scala

```diff
@@ -3,7 +3,7 @@ package symsim
 import org.scalacheck.Gen
 trait ValueFunction[State, ObservableState, Action, Reward, Scheduler[_]]:
   type VF
-  def bestAction (vf: VF) (s: State): Action
-  def chooseAction (vf: VF) (s: State): Scheduler[Action]
+  def bestAction (vf: VF) (s: ObservableState): Action
```

Comment:
This is not an issue, I just want to learn. As I know, there is no difference between State and ObservableState in ValueFunction, am I wrong?

Reply:
@mohsen-ghaffari1992, There is a difference. The value function is something that an algorithm maintains.  The algorithm can only make decisions based on ObservableState, because it does not see anything else. Generally we should aim at increasing the separation, so that we can run this code on real robotic systems, where there is no State at all, you can only see observable state ... one day.

But probably what it also means is that we should be able to remove State altogether from the VF type hierarchy. An issue lurking. Here you  go: issue #151 :)